Sources/Routing/RouteBuilder.swift

```diff
@@ -15,87 +16,118 @@ extension RouteBuilder {
         method: Method = .get,
         path: [String] = [],
         responder: @escaping RouteHandler
-    ) {
+        ) {
         let re = Request.Handler { try responder($0).makeResponse() }
         let path = path.pathComponents
         register(host: host, method: method, path: path, responder: re)
     }
-
+    
     public func register(method: Method = .get, path: [String] = [], responder: Responder) {
         let path = path.pathComponents
         register(host: nil, method: method, path: path, responder: responder)
     }
+    
+}
 
+/// Default implementation of `register(...metadata: [String: Any], ...)`
+/// is simply a wrapper around the original `register` function to avoid breaking changes.
+extension RouteBuilder {
+    public func register(
+        host: String? = nil,
+        method: Method = .get,
+        path: [String] = [],
+        metadata: [String: Any],
+        responder: @escaping RouteHandler
+        ) {
+        let re = Request.Handler { try responder($0).makeResponse() }
+        let path = path.pathComponents
+        register(host: host, method: method, path: path, metadata: metadata, responder: re)
+    }
+    
+    public func register(
+        host: String? = nil,
+        method: Method = .get,
+        path: [String] = [],
+        metadata: [String: Any],
+        responder: Responder
+        ) {
+        let path = path.pathComponents
+        register(host: host, method: method, path: path, responder: responder)
+    }
+    
 }
 
 extension RouteBuilder {
     #if swift(>=4)
     public func add(
         _ method: HTTP.Method,
         _ path: String ...,
+        metadata: [String: Any] = [:],
```

Comment:
I'd make this addition of a parameter a second declaration, rather than the a default value. So you'd end up with 2 functions, one with and one without the metadata parameter.

The function without metadata could then call the function _with_ metadata with an empty dictionary for an argument.

Reply:
@Joannis does the default implementation change your thoughts here?